### ISSUE TITLE
Changing learning rate mid training

### ISSUE THREAD

**Issue by nghiemvdv** [facebookresearch/detr#301]
I'm training a model with the default learning rate (--lr=0.0001) and customized the main.py script to save checkpoint every 5 epochs.
Now my loss and class_error curves are going up after about 120 epochs.
So I tried to load checkpoint of the 110th epoch (checkpoint0109.pth) and lowered the learning rate to 1e-05 (--lr=0.00001) then continued training from there.
Apparently, the lr does not change due to line 180 in `main.py` script:

https://github.com/facebookresearch/detr/blob/a54b77800eb8e64e3ad0d8237789fcbf2f8350c5/main.py#L179-L182

Is this expected behavior, that changing lr while loading a previous checkpoint is not intended/recommended?
or I can delete the optimizer after loading checkpoint, something like this:
```
del checkpoint['optimizer']
del checkpoint['lr_scheduler']
```
while still keep the epoch number of course.
browsing the issues did not help me or maybe I missed something
also this is the first time I post an issue, hopefully I'm not missing any detail.

**Comment by Mahbubalam4545** [comment]
The easy bit would be:
1. At first, train the model without '--resume' command line argument for 110 epochs and stop training;
2. Change the default learning rate as you intend to;
3. Now, train the model with  '--resume [path of checkpoint0109.pth]'

**Comment by alcinos** [comment]
Hi @nghiemvdv 
I definitely don't recommend deleting the optimizer, as you will loose the adam state and thus damage your optimization.

You can force the learning rate you want by putting something like this after the loading:
```
optimizer.param_groups[0]["lr"] = 0.1 # lr you want for the transformer
optimizer.param_groups[1]["lr"] = 0.01 # lr you want for the backbone
```

I believe I have answered your question and as such  I'm closing this. Feel free to reach out if you have further concerns.

**Comment by nghiemvdv** [comment]
I think it worked, but I have another question.
Do I have to initialize the ```lr_scheduler``` again because I change the learning rate after the fact, something like:
```
optimizer.param_groups[0]['lr'] = args.lr
optimizer.param_groups[1]['lr'] = args.lr_backbone
lr_scheduler = torch.optim.lr_scheduler.StepLR(optimizer, args.lr_drop)
```
right after loading state dict?
